docs/adr/adr-0022-new-icon-system.md

```diff
@@ -0,0 +1,232 @@
+```html
+<cc-icon role="img" aria-label="Caution">
+  <svg aria-hidden="true">...</svg>
+</cc-icon>
+```
+
+Unfortunately, testing this implementation in real life showed that informative icons were ignored by VoiceOver on MacOS and iOS.
+
+Using a `role="img"` and `aria-label` attributes on a custom element does not seem to be supported on Safari.
+
+#### Using the `<svg>` element itself
+
+Since we could not rely on the custom element itself, we had to rely on the `<svg>` element and manipulate it depending on its context.
+
+If the developer does not provide any `accessible-name` prop, the component adds an `aria-hidden="true"` attribute to the inner `<svg>` element.
+If the developer does provide an `accessible-name` prop, the component adds to the inner `<svg>` element:
+
+* A `role="img"` attribute,
+* An `aria-label` which value matches the value of the `accessible-name` prop,
+* A `<title>` element which value matches the value of the `accessible-name` prop.
+
+The addition of the `<title>` element is not necessary but this is the native HTML way for informative `<svg>` elements so we decided to keep it.
+
+Example of an informative `<cc-icon>`:
+```html
+<cc-icon accessible-name="Caution">
+  <svg role="img" aria-label="Caution">
+    <title>Caution</title>
+    ...
+  </svg>
+<cc-icon>
+```
+
+#### Deciding on prop names
+
+We discussed several options about the component API and how developers should use it:
+
+##### Using ARIA attributes directly
+
+This option was only possible if we used the `<cc-icon>` element itself (see "Using the `<svg>` element itself" above).
+
+Developers would directly add `aria-hidden=true` or `role="img"` + `aria-label` to the `<cc-icon>` element itself.
+
+We did not choose to go this way because:
+
+* There was too much room for mistakes. Developers could add the `role="img"` attribute and forget to add the `aria-label` attribute.
+* Relying on the `<cc-icon>` element itself and not the `<svg>` does not work on Safari.
+
+##### Reproducing the img API
+
+We considered using an `alt` prop and attribute to mimic the `<img>` API.
+
+We did not choose to go this way because we want to avoid using existing native attributes like `title` for instance.
+
+##### Using a custom accessible-name attribute
+
+We chose to use a `accessible-name` attribute that should only be used if the icon conveys information.
+
+This simplifies the API for developers because:
+
+* In most cases, icons are decorative so there is nothing to do.
+* When the icon is informative, they only have one attribute to add.
+* We already use an `accessible-name` attribute on the `<cc-button>`. This means this attribute should be familiar for developers using our components. It is also a well-known concept for many people who have worked on accessibility before.
+
+## Limits and prospects
+
+Where Remix Icon is a great library, it is currently on [pause](https://github.com/Remix-Design/RemixIcon/issues/232).
```

Comment:
He seems to be back on stage now: https://github.com/Remix-Design/RemixIcon/issues/232#issuecomment-1017140063

Reply:
Yes! Unfortunately, this message is one year old and there hasn't been much activity since then. :shrug:
I'll slightly update the sentence nevertheless.